lib/init/grass.py

```diff
@@ -1837,8 +1837,11 @@ def sh_like_startup(location, location_name, grass_env_file, sh):
     # bash histroy file handled in specific_addition
     if not sh == "bash":
         os.environ['HISTFILE'] = os.path.join(location, sh_history)
-    if not os.getenv('HISTSIZE') and not os.getenv('HISTFILESIZE'):
-        os.environ['HISTSIZE'] = "3000"
+
+    if sh == "bash":
+        # set bash history to record an unlimited command history
+        os.environ['HISTSIZE'] = "-1"
+        os.environ['HISTFILESIZE'] = os.getenv('HISTSIZE')
```

Comment:
Hardcoding the same value or saving in the a (Python) variable seems to me as more readable alternative to getting from the environment. (It might be a insignificantly faster too.) The original code might have been just some legacy from shell to Python porting.